develop branch fails to build with STRUMPACK enabled

Linking HiOp library to STRUMPACK fails with error output:
```shell
[ 40%] Linking CXX shared library libhiop.so
/usr/bin/ld: /qfs/projects/exasgd/src/spack/opt/spack/linux-rhel7-power9le/gcc-7.4.0/strumpack-5.1.1-x55sdhvsvruzl6c7yif2hz6akb7ufpgy/lib/libstrumpack.a(FrontFactory.cpp.o): In function `strumpack::FrontalMatrixBLR<std::complex<float>, long long>::partition(strumpack::SPOptions<std::complex<float> > const&, strumpack::CompressedSparseMatrix<std::complex<float>, long long> const&, long long*, bool, int)':
/scratch/manc568/exasgd-spack-stage/spack-stage-strumpack-5.1.1-x55sdhvsvruzl6c7yif2hz6akb7ufpgy/spack-src/src/sparse/fronts/FrontalMatrixBLR.hpp:497:(.text._ZN9strumpack16FrontalMatrixBLRISt7complexIfExE9partitionERKNS_9SPOptionsIS2_EERKNS_22CompressedSparseMatrixIS2_xEEPxbi[_ZN9strumpack16FrontalMatrixBLRISt7complexIfExE9partitionERKNS_9SPOptionsIS2_EERKNS_22CompressedSparseMatrixIS2_xEEPxbi]+0x2c8): toc optimization is not supported for 0xe4090003 instruction.
/usr/bin/ld: /qfs/projects/exasgd/src/spack/opt/spack/linux-rhel7-power9le/gcc-7.4.0/strumpack-5.1.1-x55sdhvsvruzl6c7yif2hz6akb7ufpgy/lib/libstrumpack.a(HODLRMatrix.cpp.o): In function `strumpack::HODLR::HODLRMatrix<float>::factor_memory() const':
/scratch/manc568/exasgd-spack-stage/spack-stage-strumpack-5.1.1-x55sdhvsvruzl6c7yif2hz6akb7ufpgy/spack-src/src/HODLR/HODLRMatrix.hpp:300:(.text._ZNK9strumpack5HODLR11HODLRMatrixIfE19total_factor_memoryEv[_ZNK9strumpack5HODLR11HODLRMatrixIfE19total_factor_memoryEv]+0x68): toc optimization is not supported for 0xe7e90002 instruction.
/usr/bin/ld: /qfs/projects/exasgd/src/spack/opt/spack/linux-rhel7-power9le/gcc-7.4.0/strumpack-5.1.1-x55sdhvsvruzl6c7yif2hz6akb7ufpgy/lib/libstrumpack.a(HODLRMatrix.cpp.o): In function `strumpack::HODLR::HODLRMatrix<double>::factor_memory() const':
/scratch/manc568/exasgd-spack-stage/spack-stage-strumpack-5.1.1-x55sdhvsvruzl6c7yif2hz6akb7ufpgy/spack-src/src/HODLR/HODLRMatrix.hpp:300:(.text._ZNK9strumpack5HODLR11HODLRMatrixIdE19total_factor_memoryEv[_ZNK9strumpack5HODLR11HODLRMatrixIdE19total_factor_memoryEv]+0x68): toc optimization is not supported for 0xe7e90002 instruction.
/usr/bin/ld: /qfs/projects/exasgd/src/spack/opt/spack/linux-rhel7-power9le/gcc-7.4.0/strumpack-5.1.1-x55sdhvsvruzl6c7yif2hz6akb7ufpgy/lib/libstrumpack.a(HODLRMatrix.cpp.o): In function `strumpack::HODLR::HODLRMatrix<std::complex<float> >::factor_memory() const':
/scratch/manc568/exasgd-spack-stage/spack-stage-strumpack-5.1.1-x55sdhvsvruzl6c7yif2hz6akb7ufpgy/spack-src/src/HODLR/HODLRMatrix.hpp:300:(.text._ZNK9strumpack5HODLR11HODLRMatrixISt7complexIfEE19total_factor_memoryEv[_ZNK9strumpack5HODLR11HODLRMatrixISt7complexIfEE19total_factor_memoryEv]+0x68): toc optimization is not supported for 0xe7e90002 instruction.
/usr/bin/ld: /qfs/projects/exasgd/src/spack/opt/spack/linux-rhel7-power9le/gcc-7.4.0/strumpack-5.1.1-x55sdhvsvruzl6c7yif2hz6akb7ufpgy/lib/libstrumpack.a(HODLRMatrix.cpp.o): In function `strumpack::HODLR::HODLRMatrix<std::complex<double> >::factor_memory() const':
/scratch/manc568/exasgd-spack-stage/spack-stage-strumpack-5.1.1-x55sdhvsvruzl6c7yif2hz6akb7ufpgy/spack-src/src/HODLR/HODLRMatrix.hpp:300:(.text._ZNK9strumpack5HODLR11HODLRMatrixISt7complexIdEE19total_factor_memoryEv[_ZNK9strumpack5HODLR11HODLRMatrixISt7complexIdEE19total_factor_memoryEv]+0x68): toc optimization is not supported for 0xe7e90002 instruction.
/usr/bin/ld: /qfs/projects/exasgd/src/spack/opt/spack/linux-rhel7-power9le/gcc-7.4.0/strumpack-5.1.1-x55sdhvsvruzl6c7yif2hz6akb7ufpgy/lib/libstrumpack.a(TaskTimer.cpp.o): In function `void __gnu_cxx::new_allocator<std::_List_node<strumpack::TaskTimer> >::destroy<strumpack::TaskTimer>(strumpack::TaskTimer*)':
/qfs/projects/ops/rh7p9/gcc/7.4.0/include/c++/7.4.0/ext/new_allocator.h:140:(.text._ZNSt7__cxx1110_List_baseIN9strumpack9TaskTimerESaIS2_EE8_M_clearEv[_ZNSt7__cxx1110_List_baseIN9strumpack9TaskTimerESaIS2_EE8_M_clearEv]+0x48): call to `strumpack::TaskTimer::~TaskTimer()' lacks nop, can't restore toc; recompile with -fPIC
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
make[2]: *** [src/libhiop.so] Error 1
make[1]: *** [src/CMakeFiles/hiop_shared.dir/all] Error 2
make: *** [all] Error 2
```
Perhaps a clue for this issue comes from the CMake warning when generating make files:
```shell
CMake Error: install(EXPORT "hiop-targets" ...) includes target "hiop_tpl" which requires target "STRUMPACK" that is
 not in any export set.
 Generating done
```
The issue can be reproduced on Newell with build scripts and modules there.

--- a/cmake/FindHiopSTRUMPACK.cmake
+++ b/cmake/FindHiopSTRUMPACK.cmake
@@ -27,6 +27,7 @@ if(STRUMPACK_LIBRARIES)
   message(STATUS "Found STRUMPACK library: ${STRUMPACK_LIBRARIES}")
   add_library(STRUMPACK INTERFACE)
   target_link_libraries(STRUMPACK INTERFACE STRUMPACK::strumpack)
+  install(TARGETS STRUMPACK EXPORT hiop-targets)
 
   # ignore SCALAPACK not_found: it may be that strumpack was built without MPI/SCALAPCK
   if(SCALAPACK_LIBRARY)

